cwms-data-api/src/test/java/cwms/cda/data/dao/watersupply/WaterContractDaoTestIT.java

```diff
@@ -0,0 +1,386 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.data.dao.watersupply;
+
+import cwms.cda.api.DataApiTestIT;
+import cwms.cda.api.enums.Nation;
+import cwms.cda.data.dao.DeleteRule;
+import cwms.cda.data.dao.LocationsDaoImpl;
+import cwms.cda.data.dao.project.ProjectDao;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.Location;
+import cwms.cda.data.dto.LookupType;
+import cwms.cda.data.dto.project.Project;
+import cwms.cda.data.dto.watersupply.PumpType;
+import cwms.cda.data.dto.watersupply.WaterSupplyPump;
+import cwms.cda.data.dto.watersupply.WaterUser;
+import cwms.cda.data.dto.watersupply.WaterUserContract;
+import cwms.cda.helpers.DTOMatch;
+import fixtures.CwmsDataApiSetupCallback;
+import mil.army.usace.hec.test.database.CwmsDatabaseContainer;
+import org.jooq.DSLContext;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import static cwms.cda.data.dao.JooqDao.getDslContext;
+import static org.junit.jupiter.api.Assertions.*;
+
+class WaterContractDaoTestIT extends DataApiTestIT {
+    private static final String OFFICE_ID = "SPK";
+    private static final String DELETE_ACTION = "DELETE ALL";
+    private static final Location testLocation = buildTestLocation("Test Location Name", "Test Location");
+    private static final Project testProject = buildTestProject();
+    private static final WaterUser testUser = buildTestWaterUser("Test User");
+
+    @BeforeAll
+    static void setup() throws Exception {
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            LocationsDaoImpl dao = new LocationsDaoImpl(ctx);
+            ProjectDao projectDao = new ProjectDao(ctx);
+            try {
+                projectDao.create(testProject);
+                dao.storeLocation(testLocation);
+            } catch (IOException e) {
+                throw new RuntimeException("Failed to store location or project", e);
+            }
+        });
+    }
+
+    @AfterAll
+    static void tearDown() throws Exception {
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            ProjectDao projectDao = new ProjectDao(ctx);
+            projectDao.delete(OFFICE_ID, testProject.getLocation().getName(), DeleteRule.DELETE_ALL);
```

Comment:
This cleans up all related water supply data and the location?

Reply:
Added location cleanup, confirmed contract and water user cleanup are run at end of each test